pkg/client/cache/fifo.go

```diff
@@ -46,7 +48,15 @@ func (f *FIFO) Add(obj interface{}) error {
 	}
 	f.lock.Lock()
 	defer f.lock.Unlock()
-	if _, exists := f.items[id]; !exists {
+	if oldObject, exists := f.items[id]; !exists {
+		// reject attempts to add older entries
+		if older, err := f.isOlder(obj, oldObject); err != nil {
```

Comment:
would prefer to accept if isNewer so we can avoid the no-op acceptance of identical items.  (or reject if isOlderOrEqual, but that's clunky)